File: solutions/numbers/Fibonacci_Sequence.js
#!/usr/bin/env node

/**
 * https://github.com/mkenney/Projects
 * Enter a number and have the program generate the Fibonacci sequence to that
 * number or to the Nth number.
 */

var readline = require('readline');
var iterations = 0;

if (process.argv.length < 3) {
	prompt("How many Fibonacci numbers do you want? ", function(iterations) {
		calc(iterations);
	});
} else {
	iterations = Number(process.argv[2]);
	calc(iterations);
}

function prompt(question, callback) {
	var iface = readline.createInterface({
		input: process.stdin,
		output: process.stdout
	});
	iface.question(question, function(answer) {
		iface.close();
		callback(answer);
	});
};

function calc(iterations) {
	var cur = 1;
	var last = 1;

	for (var a = 0; a < iterations; a++) {
		console.log(cur);
		var tmp = cur;
		cur = cur + last;
		last = tmp;
	}
};

File: solutions/numbers/Fibonacci_Sequence.py
#!/usr/bin/env python

"""
https://github.com/mkenney/Projects
Enter a number and have the program generate the Fibonacci sequence to that
number or to the Nth number.
"""

import sys

if (len(sys.argv) < 2):
	iterations = int(input("How many Fibonacci numbers do you want? "))
else:
	iterations = int(sys.argv[1])

cur = 1;
last = 1;

for a in range(0, iterations):
	print cur
	tmp = cur
	cur = cur + last
	last = tmp
